### ISSUE TITLE
Minimize amount of clicks when updating extension

### ISSUE THREAD

**Issue by frankrolf** [robofont-mechanic/mechanic-2#16]
Currently an extension update through Mechanic 2 requires
→ click on `Update`
→ click on `Re-Install`
→ click on `OK, this extension has been installed`.

Can this workflow be slightly streamlined?

**Comment by typemytype** [comment]
Mechanic does not check for updates on start up as it take some time... 
Both Jack and I have looked into threading this without succes

**Comment by jackjennings** [comment]
I think that Frank is mentioning the number of dialogs that have to be clicked through to complete an update. At one point I think the two of us discussed being able to suppress some of the default RoboFont messages, but I don't remember the result of that conversation.

**Comment by typemytype** [comment]
Mechanic only shows the standard dialogs when only 1 extension is installed, [see](https://github.com/robofont-mechanic/mechanic-2/blob/681bf76ae96c03e26a68d7dffb19220f6b9e1f75/Mechanic2.roboFontExt/lib/mechanic2/ui/controller.py#L259)

That one could also be suppressed as the user already clicked on 'install'.

**Comment by typemytype** [comment]
Frank could you elaborate where it could be streamlined?

with this limitation: mechanic does not want to request any external .json automatically...

**Comment by frankrolf** [comment]
Jack understood what I was trying to say. I simply don’t want to click three times to update an extension.

**Comment by jackjennings** [comment]
The second prompt confirming Re-Install could be removed, given that it asks the same thing as the initial click on the “Update” button.

It might also be nice to change the notification that an extension has been installed to be a notification that appears in the upper right and is dismissed automatically.

**Comment by typemytype** [comment]
like this:

```python
import AppKit

n = AppKit.NSUserNotification

notification = AppKit.NSUserNotification.alloc().init()
notification.setTitle_("hello world!")
notification.setInformativeText_("foo ")

AppKit.NSUserNotificationCenter.defaultUserNotificationCenter().deliverNotification_(notification)
```

**Comment by jackjennings** [comment]
Yeah, exactly. I'd actually argue this is a better default for all `this extension has been installed` notifications.

**Comment by verbosus** [comment]
Once either https://github.com/robofont-mechanic/mechanic-2/pull/18 or some other sort of background-thread URL-loading mechanism makes it in, how about looking into checking for updates right at Mechanic2 startup, since the UI is no longer being blocked? That would reduce at least one of Frank’s clicks. 

And I think we can get rid of the confirmation sheet “this extension has been updated,” but rather than putting it into an auto-dismissing sheet (if I understand what @typemytype and @jackjennings are proposing above) how about putting that information right into the NSCell, in the same spot where we write “Found update x → y?” 

These two changes would bring @frankrolf’s total number of clicks to update an extension down from three to one.

**Comment by typemytype** [comment]
When updates are checked in the background without blocking the main thread, Im super fine with auto check once a day (or optionally with a pref) for updates and report back if updates are available. 

I dont like auto-updates (maybe with a pref) but this can heavily interfere with user setup.

so:

* check for update on startup, without blocking the UI, once a day
* if updates are found --> pop up mechanic ui
* user has to click "update all extensions" or dismiss by closing the window

"this extension has been updated" is already converted to a notification and can be set by the users in the OS, no message dialog anymore

--> the total sum of clicks should be `1`
